update/Pos/Arbitrary/Update/Poll.hs

```diff
@@ -25,45 +24,45 @@ import           Pos.Update.Poll.Types (BlockVersionState (..), ConfirmedProposa
                                         ProposalState (..), USUndo, UndecidedProposalState (..),
                                         UpsExtra (..))
 
-instance Arbitrary HeaderHash => Arbitrary UpsExtra where
+instance Arbitrary UpsExtra where
     arbitrary = genericArbitrary
     shrink = genericShrink
 
-instance (Arbitrary HeaderHash, HasProtocolConstants, HasProtocolMagic) => Arbitrary UndecidedProposalState where
+instance (HasProtocolConstants, HasProtocolMagic) => Arbitrary UndecidedProposalState where
     arbitrary = genericArbitrary
     shrink = genericShrink
 
-instance Arbitrary HeaderHash => Arbitrary DpsExtra where
+instance Arbitrary DpsExtra where
     arbitrary = genericArbitrary
     shrink = genericShrink
 
-instance (Arbitrary HeaderHash, HasProtocolConstants, HasProtocolMagic) => Arbitrary DecidedProposalState where
+instance (HasProtocolConstants, HasProtocolMagic) => Arbitrary DecidedProposalState where
     arbitrary = genericArbitrary
     shrink = genericShrink
 
-instance (Arbitrary HeaderHash, HasProtocolMagic) => Arbitrary ConfirmedProposalState where
+instance (HasProtocolMagic) => Arbitrary ConfirmedProposalState where
     arbitrary = genericArbitrary
     shrink = genericShrink
 
-instance (Arbitrary HeaderHash, HasProtocolConstants, HasProtocolMagic) => Arbitrary ProposalState  where
+instance (HasProtocolConstants, HasProtocolMagic) => Arbitrary ProposalState  where
     arbitrary = genericArbitrary
     shrink = genericShrink
 
-instance Arbitrary HeaderHash => Arbitrary BlockVersionState where
+instance Arbitrary BlockVersionState where
     arbitrary = genericArbitrary
     shrink = genericShrink
 
-instance (Arbitrary HeaderHash, HasProtocolConstants, HasProtocolMagic) => Arbitrary PollModifier where
+instance (HasProtocolConstants, HasProtocolMagic) => Arbitrary PollModifier where
     arbitrary = genericArbitrary
     shrink = genericShrink
 
-instance (Arbitrary HeaderHash, HasProtocolConstants, HasProtocolMagic) => Arbitrary PollState where
+instance (HasProtocolConstants, HasProtocolMagic) => Arbitrary PollState where
     arbitrary = do
         ps <- genericArbitrary
         return (ps & psActivePropsIdx %~ HM.filter (not . null))
     shrink = genericShrink
 
-instance (Arbitrary HeaderHash, HasProtocolConstants, HasProtocolMagic) => Arbitrary USUndo where
+instance (HasProtocolConstants, HasProtocolMagic) => Arbitrary USUndo where
```

Comment:
It's shocking to see so many instance constraints like `Arbitrary HeaderHash`. All these do is defer the "missing constraint" error to use sites. Is there a reason that they're there?

Reply:
I've no idea. They've been caught now thanks to the new GHC version.